src/github.ts

```diff
@@ -1015,52 +1015,151 @@ export class GitHub {
+   * the provided prefix.
+   *
+   * If a ref is provided, consider the list of files at that
+   * sha. Otherwise, use the default branch.
+   *
+   * @param filename The name of the file to find
+   * @param prefix Optional path prefix used to filter results
+   * @param ref Optional git reference. Defaults to the default
+   *   branch's HEAD.
+   */
+  async findFilesByFilenameAndRef(
     filename: string,
+    ref: string,
     prefix?: string
   ): Promise<string[]> {
-    let q = `filename:${filename}+repo:${this.owner}/${this.repo}`;
     if (prefix) {
       prefix = this.normalizePrefix(prefix);
-      q += `+path:${prefix}`;
     }
-    const response: {data: FileSearchResponse} = await this.octokit.search.code(
-      {
-        q: q,
-      }
-    );
-    return response.data.items.map(file => {
-      let path = file.path;
-      if (prefix) {
-        const pfix = new RegExp(`^${prefix}[/\\\\]`);
-        path = path.replace(pfix, '');
-      }
-      return path;
+    const response: {
+      data: GitGetTreeResponse;
+    } = await this.octokit.git.getTree({
+      owner: this.owner,
+      repo: this.repo,
+      tree_sha: ref,
+      recursive: 'true',
     });
+    return response.data.tree
+      .filter(file => {
+        const path = file.path;
+        return (
+          path &&
+          // match the filename
+          path.endsWith(filename) &&
+          // match the prefix if provided
+          (!prefix || path.startsWith(prefix))
+        );
+      })
+      .map(file => {
+        let path = file.path!;
+        // strip the prefix if provided
+        if (prefix) {
+          const pfix = new RegExp(`^${prefix}[/\\\\]`);
+          path = path.replace(pfix, '');
+        }
+        return path;
+      });
+  }
+  /**
+   * Returns a list of paths to all files with a given name.
+   *
+   * If a prefix is specified, only return paths that match
+   * the provided prefix.
+   *
+   * @param filename The name of the file to find
+   * @param prefix Optional path prefix used to filter results
+   */
+  async findFilesByFilename(
+    filename: string,
+    prefix?: string
+  ): Promise<string[]> {
+    return this.findFilesByFilenameAndRef(
+      filename,
+      await this.getDefaultBranch(),
+      prefix
+    );
   }
 
-  async findFilesByExtension(
+  /**
+   * Returns a list of paths to all files with a given file
+   * extension.
+   *
+   * If a prefix is specified, only return paths that match
+   * the provided prefix.
+   *
+   * If a ref is provided, consider the list of files at that
+   * sha. Otherwise, use the default branch.
+   *
+   * @param extension The file extension used to filter results.
+   *   Example: `js`, `java`
+   * @param prefix Optional path prefix used to filter results
+   * @param ref Optional git reference. Defaults to the default
+   *   branch's HEAD.
+   */
+  async findFilesByExtensionAndRef(
     extension: string,
+    ref: string,
```

Comment:
Should ref be optional, based on the description of the method?

Reply:
I originally had it optional as an overload of `fileFilesByExtension(extension, prefix?, ref?)`, but then opted to make it a separate explicit method. I'll fix the docs to not make it optional.

Reply:
Done